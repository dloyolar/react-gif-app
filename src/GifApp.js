import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({ defaultCategories = ['Haikyuu'] }) => {
  // const [categories, setCategories] = useState(['Kimetsu no Yaiba']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
