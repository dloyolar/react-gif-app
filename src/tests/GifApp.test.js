import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { GifApp } from '../GifApp';

describe('test <GifApp />', () => {
  test('should mount <GifApp />', () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = ['One Punch', 'Haikyuu'];

    const wrapper = shallow(<GifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
