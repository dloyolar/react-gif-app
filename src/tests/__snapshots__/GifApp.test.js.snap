// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifApp /> should mount <GifApp /> 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Haikyuu"
      key="Haikyuu"
    />
  </ol>
</Fragment>
`;

exports[`test <GifApp /> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Haikyuu"
      key="Haikyuu"
    />
  </ol>
</Fragment>
`;
